Rubberduck.RibbonDispatcher/Abstract/RibbonCommon.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using stdole;
+using Microsoft.Office.Core;
+
+using Rubberduck.RibbonDispatcher;
+using Rubberduck.RibbonDispatcher.Abstract;
+
+namespace Rubberduck.RibbonDispatcher.Abstract
+{
+    using ControlSize         = RibbonControlSize;
+    using LanguageStrings     = IRibbonTextLanguageControl;
+    using ChangedEventHandler = EventHandler<ChangedControlEventArgs>;
+
+    public abstract class RibbonCommon : IRibbonCommon {
+        protected RibbonCommon(string id, LanguageStrings strings, bool visible, bool enabled, ControlSize size) {
+            Id               = id;
+            _languageStrings = strings;
+            _visible         = visible;
+            _enabled         = enabled;
+            _size            = size;
+        }
+
+        public event ChangedEventHandler Changed;
+
+        public string       Id          { get; }
+
+        public string       Description => _languageStrings?.Description??"";
+        public string       KeyTip      => _languageStrings?.KeyTip??"";
+        public string       Label       => Use2ndLabel ? _languageStrings?.AlternateLabel??Id 
+                                                       : _languageStrings.Label??Id;
+        public string       ScreenTip   => _languageStrings?.ScreenTip??Id;
+        public string       SuperTip    => _languageStrings?.SuperTip??"";
+
+        public bool         Enabled     { get {return _enabled;}     set {_enabled     = value; OnChanged();} } private bool         _enabled;
+        public IPictureDisp Image       { get {return _image;}       set {_image       = value; OnChanged();} } private IPictureDisp _image;
+        public ControlSize  Size        { get {return _size;}        set {_size        = value; OnChanged();} } private ControlSize  _size;
+        public bool         Use2ndLabel { get {return _use2ndLabel;} set {_use2ndLabel = value; OnChanged();} } private bool         _use2ndLabel;
+        public bool         Visible     { get {return _visible;}     set {_visible     = value; OnChanged();} } private bool         _visible;
```

Comment:
this feels incredibly unreadable to me

Reply:
This is the one point where I think I might push back hard. I spend a lot of effort making really terrible copy-paste, cargo-cult code written by Financial Analysts recognizable, never mind readable. One thing I have learned is that emphasizing all possible patterns in repetitive code, so that the "sameness" is completely obvious, pays immense dividends. I agree that there is an ugliness to this code; yet the "sameness" of the pattern is immediately obvious now, and will still be in 6 months or 18 months. One look affirms that all the properties are following the same pattern, so only one needs to be looked at closely.